### DISCUSSION TITLE
deploy to Vercel error (Build error)

### DISCUSSION THREAD

**Discussion by rhmndhika** [vercel/next.js#19783] [Help] [answered]
Hi, i'm new to github and tried to build my website with vercel. i had a problem with deployment in vercel.

this is my github https://github.com/rhmndhika/Personal-Web.
And this is the log from deployment with vercel : 

18:21:26 | Cloning github.com/rhmndhika/Personal-Web (Branch: main, Commit: d0a5d14)
-- | --
18:21:27 | Cloning completed in 1132ms
18:21:27 | Analyzing source code...
18:21:27 | Installing build runtime...
18:21:28 | Build runtime installed: 404.257ms
18:21:28 | Looking up build cache...
18:21:28 | Build cache not found
18:21:29 | Installing dependencies...
18:21:51 | > @fortawesome/fontawesome-common-types@0.2.32 postinstall /vercel/workpath0/node_modules/@fortawesome/fontawesome-common-types
18:21:51 | > node attribution.js
18:21:51 | Font Awesome Free 0.2.32 by @fontawesome - https://fontawesome.com
18:21:51 | License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
18:21:51 | > core-js@2.6.12 postinstall /vercel/workpath0/node_modules/babel-runtime/node_modules/core-js
18:21:51 | > node -e "try{require('./postinstall')}catch(e){}"
18:21:51 | > core-js@3.8.0 postinstall /vercel/workpath0/node_modules/core-js
18:21:51 | > node -e "try{require('./postinstall')}catch(e){}"
18:21:51 | > core-js-pure@3.8.0 postinstall /vercel/workpath0/node_modules/core-js-pure
18:21:51 | > node -e "try{require('./postinstall')}catch(e){}"
18:21:51 | > ejs@2.7.4 postinstall /vercel/workpath0/node_modules/ejs
18:21:51 | > node ./postinstall.js
18:21:52 | > @fortawesome/fontawesome-svg-core@1.2.32 postinstall /vercel/workpath0/node_modules/@fortawesome/fontawesome-svg-core
18:21:52 | > node attribution.js
18:21:52 | Font Awesome Free 1.2.32 by @fontawesome - https://fontawesome.com
18:21:52 | License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
18:21:52 | > @fortawesome/free-brands-svg-icons@5.15.1 postinstall /vercel/workpath0/node_modules/@fortawesome/free-brands-svg-icons
18:21:52 | > node attribution.js
18:21:52 | Font Awesome Free 5.15.1 by @fontawesome - https://fontawesome.com
18:21:52 | License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License)
18:21:53 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents):
18:21:53 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
18:21:53 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/watchpack-chokidar2/node_modules/fsevents):
18:21:53 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
18:21:53 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.2.1 (node_modules/fsevents):
18:21:53 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.2.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
18:21:53 | npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/chokidar/node_modules/fsevents):
18:21:53 | npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
18:21:53 | added 1980 packages from 819 contributors in 24.17s
18:21:54 | 124 packages are looking for funding
18:21:54 | run `npm fund` for details
18:21:55 | Running "npm run build"
18:21:55 | > aries-web@0.1.0 build /vercel/workpath0
18:21:55 | > react-scripts build
18:21:56 | Creating an optimized production build...
18:22:07 | Treating warnings as errors because process.env.CI = true.
18:22:07 | Most CI servers set it automatically.
18:22:07 | Failed to compile.
18:22:07 | src/Components/about.js
18:22:07 | Line 11:7:  img elements must have an alt prop, either with meaningful text, or an empty string for decorative images  jsx-a11y/alt-text
18:22:07 | src/Components/button.js
18:22:07 | Line 2:8:  'ReactDOM' is defined but never used  no-unused-vars
18:22:07 | src/Components/home.js
18:22:07 | Line 11:8:  'Buttons' is defined but never used  no-unused-vars
18:22:07 | npm ERR! code ELIFECYCLE
18:22:07 | npm ERR! errno 1
18:22:07 | npm ERR! aries-web@0.1.0 build: `react-scripts build`
18:22:07 | npm ERR! Exit status 1
18:22:07 | npm ERR!
18:22:07 | npm ERR! Failed at the aries-web@0.1.0 build script.
18:22:07 | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
18:22:07 | npm ERR! A complete log of this run can be found in:
18:22:07 | npm ERR!     /vercel/.npm/_logs/2020-12-03T11_22_07_640Z-debug.log
18:22:07 | Error: Command "npm run build" exited with 1
18:22:09 | Done with "package.json"

i dont know what to do can someone help me.

**Comment by jamesmosier** [comment, marked as answer]
Here are your errors that you should fix:

```
18:22:07 | src/Components/about.js
-- | --
18:22:07 | Line 11:7: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text
18:22:07 | src/Components/button.js
18:22:07 | Line 2:8: 'ReactDOM' is defined but never used no-unused-vars
18:22:07 | src/Components/home.js
18:22:07 | Line 11:8: 'Buttons' is defined but never used no-unused-vars
```

* `src/Components/about.js` on line 11 you should write it as `<img className="Gambar1" src={CeBayed} alt="description of image" />` where it is a self-closing `img` tag.

it looks like you already fixed the other 2 issues in `button.js` and `home.js`


**Edit** and actually it looks like you already added the `alt` tag to `about.js` so your build should succeed 🎉

**Comment by Vturnus** [comment]
Hi,
I'm having a problem deploying Nuxt project
I build it successfully on my computer.
but it can't get built, in vercel

this is the log


13:53:52.999
[log] Running with @nuxt/vercel-builder version 0.21.3
13:53:52.999
[log]  ----------------- Prepare build ----------------- 
13:53:52.999
[log] Downloading files...
13:53:53.000
[log] Working directory: /vercel/path0
13:53:53.002
[log] Using yarn
13:53:53.003
[info] Prepare build took: 3.687876 ms
13:53:53.003
[log]  ----------------- Install devDependencies ----------------- 
13:53:53.355
yarn install v1.22.10
13:53:53.367
warning package.json: No license field
13:53:53.430
warning mastering-nuxt: No license field
13:53:53.431
[1/5] Resolving packages...
13:53:53.903
[2/5] Fetching packages...
13:54:14.199
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
13:54:14.200
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
13:54:14.212
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
13:54:14.212
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
13:54:14.219
[3/5] Linking dependencies...
13:54:14.222
warning "@nuxtjs/cloudinary > cloudinary-core@2.11.4" has unmet peer dependency "lodash@>=4.0".
13:54:14.223
warning "@nuxtjs/cloudinary > cloudinary-vue@1.2.3" has unmet peer dependency "vue@^2.0.0".
13:54:14.223
warning " > @studiometa/vue-mapbox-gl@1.4.7" has incorrect peer dependency "mapbox-gl@^1.12.0".
13:54:14.223
warning " > @studiometa/vue-mapbox-gl@1.4.7" has unmet peer dependency "vue@^2.6.10".
13:54:14.223
warning " > cloudinary-vue@1.1.2" has unmet peer dependency "vue@^2.0.0".
13:54:14.224
warning "nuxt > @nuxt/components@2.1.8" has unmet peer dependency "consola@*".
13:54:14.235
warning " > vue2-leaflet@2.7.1" has unmet peer dependency "@types/leaflet@^1.5.7".
13:54:14.235
warning " > vue2-leaflet@2.7.1" has unmet peer dependency "vue@^2.5.17".
13:54:14.235
warning " > v-calendar@2.3.2" has unmet peer dependency "@popperjs/core@^2.4.0".
13:54:14.235
warning " > v-calendar@2.3.2" has unmet peer dependency "vue@^2.5.18".
13:54:14.235
warning "@nuxtjs/tailwindcss > @nuxt/postcss8 > css-loader@5.2.7" has unmet peer dependency "webpack@^4.27.0 || ^5.0.0".
13:54:14.235
warning "@nuxtjs/tailwindcss > @nuxt/postcss8 > postcss-loader@4.3.0" has unmet peer dependency "webpack@^4.0.0 || ^5.0.0".
13:54:14.235
warning " > sass-loader@10.1.1" has unmet peer dependency "webpack@^4.36.0 || ^5.0.0".
13:54:33.206
[4/5] Building fresh packages...
13:54:35.961
error /vercel/path0/node_modules/sharp: Command failed.
13:54:35.962
Exit code: 1
13:54:35.963
Command: (node install/libvips && node install/dll-copy && prebuild-install) || (node install/can-compile && node-gyp rebuild && node install/dll-copy)
13:54:35.963
Arguments: 
13:54:35.963
Directory: /vercel/path0/node_modules/sharp
13:54:35.963
Output:
13:54:35.963
sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.10.6/libvips-8.10.6-linux-x64.tar.br
13:54:35.963
/bin/sh: prebuild-install: command not found
13:54:35.963
gyp info it worked if it ends with ok
13:54:35.963
gyp info using node-gyp@5.1.0
13:54:35.963
gyp info using node@14.17.3 | linux | x64
13:54:35.963
gyp info find Python using Python version 2.7.18 found at "/usr/bin/python"
13:54:35.963
gyp http GET https://nodejs.org/download/release/v14.17.3/node-v14.17.3-headers.tar.gz
13:54:35.963
gyp http 200 https://nodejs.org/download/release/v14.17.3/node-v14.17.3-headers.tar.gz
13:54:35.963
gyp http GET https://nodejs.org/download/release/v14.17.3/SHASUMS256.txt
13:54:35.963
gyp http 200 https://nodejs.org/download/release/v14.17.3/SHASUMS256.txt
13:54:35.963
gyp info spawn /usr/bin/python
13:54:35.963
gyp info spawn args [
13:54:35.963
gyp info spawn args   '/node14/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
13:54:35.963
gyp info spawn args   'binding.gyp',
13:54:35.963
gyp info spawn args   '-f',
13:54:35.964
gyp info spawn args   'make',
13:54:35.964
gyp info spawn args   '-I',
13:54:35.964
gyp info spawn args   '/vercel/path0/node_modules_dev/sharp/build/config.gypi',
13:54:35.964
gyp info spawn args   '-I',
13:54:35.964
gyp info spawn args   '/node14/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
13:54:35.964
gyp info spawn args   '-I',
13:54:35.964
gyp info spawn args   '/vercel/.cache/node-gyp/14.17.3/include/node/common.gypi',
13:54:35.964
gyp info spawn args   '-Dlibrary=shared_library',
13:54:35.964
gyp info spawn args   '-Dvisibility=default',
13:54:35.964
gyp info spawn args   '-Dnode_root_dir=/vercel/.cache/node-gyp/14.17.3',
13:54:35.964
gyp info spawn args   '-Dnode_gyp_dir=/node14/lib/node_modules/npm/node_modules/node-gyp',
13:54:35.964
gyp info spawn args   '-Dnode_lib_file=/vercel/.cache/node-gyp/14.17.3/<(target_arch)/node.lib',
13:54:35.964
gyp info spawn args   '-Dmodule_root_dir=/vercel/path0/node_modules_dev/sharp',
13:54:35.964
gyp info spawn args   '-Dnode_engine=v8',
13:54:35.964
gyp info spawn args   '--depth=.',
13:54:35.964
gyp info spawn args   '--no-parallel',
13:54:35.964
gyp info spawn args   '--generator-output',
13:54:35.964
gyp info spawn args   'build',
13:54:35.964
gyp info spawn args   '-Goutput_dir=.'
13:54:35.965
gyp info spawn args ]
13:54:35.965
gyp info spawn make
13:54:35.965
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
13:54:35.965
make: Entering directory `/vercel/path0/node_modules_dev/sharp/build'
13:54:35.965
  CC(target) Release/obj.target/nothing/../node-addon-api/nothing.o
13:54:35.965
  AR(target) Release/obj.target/../node-addon-api/nothing.a
13:54:35.965
  COPY Release/nothing.a
13:54:35.965
  TOUCH Release/obj.target/libvips-cpp.stamp
13:54:35.965
  CXX(target) Release/obj.target/sharp/src/common.o
13:54:35.965
../src/common.cc:24:10: fatal error: vips/vips8: No such file or directory
13:54:35.965
 #include <vips/vips8>
13:54:35.965
          ^~~~~~~~~~~~
13:54:35.965
compilation terminated.
13:54:35.965
make: *** [Release/obj.target/sharp/src/common.o] Error 1
13:54:35.965
make: Leaving directory `/vercel/path0/node_modules_dev/sharp/build'
13:54:35.965
gyp ERR! build error 
13:54:35.965
gyp ERR! stack Error: `make` failed with exit code: 2
13:54:35.965
gyp ERR! stack     at ChildProcess.onExit (/node14/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
13:54:35.965
gyp ERR! stack     at ChildProcess.emit (events.js:375:28)
13:54:35.966
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)
13:54:35.966
gyp ERR! System Linux 4.14.238-182.422.amzn2.x86_64
13:54:35.966
gyp ERR! command "/node14/bin/node" "/node14/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
13:54:35.966
gyp ERR! cwd /vercel/path0/node_modules_dev/sharp
13:54:35.966
gyp ERR! node -v v14.17.3
13:54:35.966
gyp ERR! node-gyp -v v5.1.0
13:54:35.966
gyp ERR! not ok
13:54:35.966
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
13:54:36.002
Error: Command "yarn install" exited with 1
